receptorctl/noxfile.py

```diff
@@ -28,12 +29,42 @@ def install(session: nox.Session, *args, req: str, **kwargs):
     session.install("-r", requirements_directory / f"{req}.in", *args, **kwargs)
 
 
+def version(session: nox.Session):
+    """
+    Create a .VERSION file.
+    """
+    try:
+        official_version = session.run(
+            "git",
+            "describe",
+            "--exact-match",
+            "--tags",
+            external=True,
+            stderr=open(os.devnull, "w"),
+        )
+    except nox.command.CommandFailed:
+        official_version = None
+
+    if not official_version:
+        print("Current commit not tagged. Using closest annotated tag instead.")
+        tag = session.run(
+            "git", "describe", "--tags", "--always", silent=True, external=True
+        )
+        rev = session.run(
+            "git", "rev-parse", "--short", "HEAD", silent=True, external=True
+        )
+        version = tag.split("-")[0] + "+" + rev
+
+    Path(".VERSION").write_text(version)
+
+
 @nox.session(python=python_versions)
 def tests(session: nox.Session):
     """
     Run receptorctl tests
     """
     install(session, req="tests")
+    version(session)
```

Comment:
It doesn't look like this file is even used anywhere. And it's calculated from Git, calling subprocesses on every session in the matrix.
Why do you need it? Are you sure you want it as a part of the PR that simply changes the CI?

Reply:
The tests sessions fail without it. I was attempting to avoid the need to call nox from the makefile so CI can call nox directly.

In the makefile the file is created here: https://github.com/ansible/receptor/blob/dd4efd9a0b703d8cf975bcf0a3d9af5f21acfdf0/Makefile#L3

And then it gets used for receptorctl targets like this one: https://github.com/ansible/receptor/blob/dd4efd9a0b703d8cf975bcf0a3d9af5f21acfdf0/Makefile#L91

Reply:
Oh, I see. Then, I'd move this in-memory, into the pytest runs through its hooks. But maybe, later.

Reply:
Hmm. I see what you mean. I wonder if it's better to do that now or later. @thom-at-redhat What do you think?